packages/cli/command_spec.pony

```diff
@@ -0,0 +1,346 @@
+    match default'
+    | None => (typ', false, true)
+    | let d: Value => (typ', d, false)
+    else
+      // Ponyc limitation: else can't happen, but segfaults without it
+      (BoolType, false, false)
+    end
+
+  new val bool(
+    name': String,
+    descr': String = "",
+    short': (U8 | None) = None,
+    default': (Bool | None) = None)
+  =>
+    _name = name'
+    _descr = descr'
+    _short = short'
+    (_typ, _default, _required) = _init(BoolType, default')
+
+  new val string(
+    name': String,
+    descr': String = "",
+    short': (U8 | None) = None,
+    default': (String | None) = None)
+  =>
+    _name = name'
+    _descr = descr'
+    _short = short'
+    (_typ, _default, _required) = _init(StringType, default')
+
+  new val i64(name': String,
+    descr': String = "",
+    short': (U8 | None) = None,
+    default': (I64 | None) = None)
+  =>
+    _name = name'
+    _descr = descr'
+    _short = short'
+    (_typ, _default, _required) = _init(I64Type, default')
+
+  new val f64(name': String,
+    descr': String = "",
+    short': (U8 | None) = None,
+    default': (F64 | None) = None)
+  =>
+    _name = name'
+    _descr = descr'
+    _short = short'
+    (_typ, _default, _required) = _init(F64Type, default')
+
+  fun name(): String => _name
+
+  fun descr(): String => _descr
+
+  fun typ(): ValueType => _typ
+
+  fun default(): Value => _default
+
+  fun required(): Bool => _required
+
+  // Other than bools, all options require args.
+  fun _requires_arg(): Bool =>
+    match _typ
+    | let _: BoolType => false
+    else
+      true
+    end
+
+  // Used for bool options to get the true arg when option is present w/o arg
+  fun _default_arg(): Value =>
+    match _typ
+    | let _: BoolType => true
+    else
+      false
+    end
+
+  fun _has_short(sh: U8): Bool =>
+    match _short
+    | let ss: U8 => sh == ss
+    else
+      false
+    end
+
+  fun help_string(): String =>
+    let s =
+      match _short
+      | let ss: U8 => "-" + String.from_utf32(ss.u32()) + ", "
+      else
+        "    "
+      end
+    s + "--" + _name +
+      if not _required then "=" + _default.string() else "" end
+
+  fun deb_string(): String =>
+    "--" + _name + "[" + _typ.string() + "]" +
+      if not _required then "(=" + _default.string() + ")" else "" end
+
+class val ArgSpec
+  """
+  ArgSpec describes the specification of a positional Arg(ument). They have a
+  name, descr(iption), a typ(e), and a default value when they are not required.
+
+  Args always come after a leaf command, and are assigned in their positional
+  order.
+  """
+  let _name: String
+  let _descr: String
+  let _typ: ValueType
+  let _default: Value
+  let _required: Bool
+
+  fun tag _init(typ': ValueType, default': (Value | None))
+    : (ValueType, Value, Bool)
+  =>
+    match default'
+    | None => (typ', false, true)
+    | let d: Value => (typ', d, false)
+    else
+      // Ponyc limitation: else can't happen, but segfaults without it
```

Comment:
Yes, this still happens, and I meant to create a smaller repro case. I'll try to put that together, but I'll create a ticket in the meantime track it and refer to this PR as a case.